.github/workflows/ci.yml

```diff
@@ -47,3 +47,73 @@ jobs:
           architecture: "x64"
       - run: pip install flake8 flake8-print
       - run: flake8 --enable=T .
+
+  test-ubuntu:
+    needs: [lint-commit, lint-black, lint-flake8]
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8"]
+        backend: ["ray", "mpi", "dask", "multiprocessing", "python"]
+    env:
+      UNIDIST_BACKEND: ${{matrix.backend}}
+    name: test-ubuntu (backend ${{matrix.backend}}, python ${{matrix.python-version}})
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: unidist
+          environment-file: environment.yml
+          python-version: ${{matrix.python-version}}
+          channel-priority: strict
+      - name: Conda environment
+        run: |
+          conda info
+          conda list
+      - run: python unidist/cli -m pytest unidist/test/test_general.py
+      - run: python unidist/cli -m pytest unidist/test/test_task.py
+      - run: python unidist/cli -m pytest unidist/test/test_actor.py
+
+  test-windows:
+    needs: [lint-commit, lint-black, lint-flake8]
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8"]
+        backend: ["ray", "dask", "multiprocessing", "python"]
+    env:
+      UNIDIST_BACKEND: ${{matrix.backend}}
+    name: test-windows (backend ${{matrix.backend}}, python ${{matrix.python-version}})
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: unidist
+          environment-file: environment.yml
+          python-version: ${{matrix.python-version}}
+          channel-priority: strict
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly
+          auto-update-conda: true # this enable `use-only-tar-bz2` feature on Windows
+          # Miniconda setup sometimes fails because of an http error. retry
+          # it once if it fails. todo(https://github.com/conda-incubator/setup-miniconda/issues/129):
+          # get rid of this once setup-miniconda can be configured to retry
+          # setup on http errors.
+          max_attempts: 2
```

Comment:
Do we have frequent failures too?

Reply:
No, this is a proactive fix for possible failures due to connection problems on Windows

Reply:
We see such an error on linux in Modin too. Should we add restart for linux?

Reply:
Since we added this for WIndows, for Linux it would be useful too. Added